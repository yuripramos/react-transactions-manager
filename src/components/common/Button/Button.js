import React from "react";
import { bool, node, string, func, number } from "prop-types";
import { ButtonTag } from "../../../styles/objects";

function Button({
  children,
  onClick,
  isCallToAction,
  isWarning,
  type,
  disabled,
  width,
  className,
  small,
  height,
  dataTest,
  paddingSize,
  opacity,
  isNewTransfer
}) {
  return (
    <ButtonTag
      className={className}
      width={width}
      height={height}
      type={type}
      disabled={disabled}
      isCallToAction={isCallToAction}
      isWarning={isWarning}
      data-test={dataTest ? dataTest : Button.displayName}
      onClick={onClick}
      small={small}
      opacity={opacity}
      paddingSize={paddingSize}
      isNewTransfer={isNewTransfer}
    >
      {children}
    </ButtonTag>
  );
}

Button.displayName = "Button";

Button.defaultProps = {
  isCallToAction: null,
  isWarning: null,
  type: "button",
  disabled: false,
  width: null,
  height: null,
  onClick: null,
  className: null,
  small: false,
  paddingSize: null,
  isNewTransfer: null
};

Button.propTypes = {
  children: node,
  isCallToAction: bool,
  isWarning: bool,
  disabled: bool,
  type: string,
  width: number,
  onClick: func,
  className: string,
  small: bool,
  height: number,
  paddingSize: number,
  isNewTransfer: bool
};

export default Button;
