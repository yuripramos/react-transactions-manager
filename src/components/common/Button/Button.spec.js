import React from "react";
import { shallow, mount } from "enzyme";

import Button from "./Button";

describe("Button component", () => {
  it("should match snapshot", () => {
    expect(shallow(<Button>Text</Button>)).toMatchSnapshot();
  });

  it('should render an "button" tag', () => {
    expect(shallow(<Button>Text</Button>)).toMatchSnapshot();
    expect(mount(<Button>Text</Button>).find("button").length).toBe(1);
  });

  it("should render highlighted class when isCallToAction is truthy", () => {
    expect(shallow(<Button isCallToAction>Text</Button>)).toMatchSnapshot();
  });

  it("should render disabled class when disabled is truthy", () => {
    expect(shallow(<Button disabled>Text</Button>)).toMatchSnapshot();
  });

  it("should render different types", () => {
    expect(shallow(<Button type="submit">Text</Button>)).toMatchSnapshot();
  });

  it("should render different width", () => {
    expect(shallow(<Button width="100">Text</Button>)).toMatchSnapshot();
  });
});
