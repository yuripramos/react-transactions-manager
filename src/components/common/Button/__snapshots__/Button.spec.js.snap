// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<styled.button
  className={null}
  data-test="Button"
  disabled={false}
  height={null}
  isCallToAction={null}
  isNewTransfer={null}
  isWarning={null}
  onClick={null}
  paddingSize={null}
  small={false}
  type="button"
  width={null}
>
  Text
</styled.button>
`;

exports[`Button component should render an "button" tag 1`] = `
<styled.button
  className={null}
  data-test="Button"
  disabled={false}
  height={null}
  isCallToAction={null}
  isNewTransfer={null}
  isWarning={null}
  onClick={null}
  paddingSize={null}
  small={false}
  type="button"
  width={null}
>
  Text
</styled.button>
`;

exports[`Button component should render different types 1`] = `
<styled.button
  className={null}
  data-test="Button"
  disabled={false}
  height={null}
  isCallToAction={null}
  isNewTransfer={null}
  isWarning={null}
  onClick={null}
  paddingSize={null}
  small={false}
  type="submit"
  width={null}
>
  Text
</styled.button>
`;

exports[`Button component should render different widths 1`] = `
<styled.button
  className={null}
  data-test="Button"
  disabled={false}
  height={null}
  isCallToAction={null}
  isNewTransfer={null}
  isWarning={null}
  onClick={null}
  paddingSize={null}
  small={false}
  type="button"
  width="100"
>
  Text
</styled.button>
`;

exports[`Button component should render disabled class when disabled is truthy 1`] = `
<styled.button
  className={null}
  data-test="Button"
  disabled={true}
  height={null}
  isCallToAction={null}
  isNewTransfer={null}
  isWarning={null}
  onClick={null}
  paddingSize={null}
  small={false}
  type="button"
  width={null}
>
  Text
</styled.button>
`;

exports[`Button component should render highlighted class when isCallToAction is truthy 1`] = `
<styled.button
  className={null}
  data-test="Button"
  disabled={false}
  height={null}
  isCallToAction={true}
  isNewTransfer={null}
  isWarning={null}
  onClick={null}
  paddingSize={null}
  small={false}
  type="button"
  width={null}
>
  Text
</styled.button>
`;
